src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.cs

```diff
@@ -277,8 +289,52 @@ private static Glyph GetGlyph(SymbolKind kind, Accessibility? declaredAccessibil
                     }
                 }
             }
+        }
+
+        private static void AddNamesFromExistingOverloads(CSharpSyntaxContext context, SemanticModel semanticModel, ArrayBuilder<(string name, SymbolKind kind)> result, CancellationToken cancellationToken)
+        {
+            var namedType = semanticModel.GetEnclosingNamedType(context.Position, cancellationToken);
+            if (namedType is null)
+                return;
+
+            var parameterSyntax = context.LeftToken.GetAncestor(n => n.IsKind(SyntaxKind.Parameter)) as ParameterSyntax;
+            if (parameterSyntax is not { Type: { } parameterType, Parent: { Parent: BaseMethodDeclarationSyntax baseMethod } })
+                return;
+
+            var methodParameterType = semanticModel.GetTypeInfo(parameterType, cancellationToken).Type;
+            if (methodParameterType is null)
+                return;
+
+            var overloads = GetOverloads(namedType, baseMethod);
+            if (overloads.IsEmpty)
+                return;
+
+            var currentParameterNames = baseMethod.ParameterList.Parameters.Select(p => p.Identifier.ValueText).ToImmutableHashSet();
+
+            foreach (var overload in overloads)
+            {
+                foreach (var overloadParameter in overload.Parameters)
+                {
+                    if (!currentParameterNames.Contains(overloadParameter.Name) &&
+                        methodParameterType.Equals(overloadParameter.Type, SymbolEqualityComparer.Default))
+                    {
+                        result.Add((overloadParameter.Name, SymbolKind.Parameter));
+                    }
+                }
+            }
+
+            return;
 
-            return result.ToImmutable();
+            // Local functions
+            static ImmutableArray<IMethodSymbol> GetOverloads(INamedTypeSymbol namedType, BaseMethodDeclarationSyntax baseMethod)
+            {
+                return baseMethod switch
+                {
+                    MethodDeclarationSyntax method => namedType.GetMembers(method.Identifier.ValueText).OfType<IMethodSymbol>().ToImmutableArray(),
```

Comment:
do you want to collect from teh base type as well?

Reply:
I haven't thought about it. I think `GetMembers` will return from base types, which isn't a bad thing. Will need to add a test confirming this.

Reply:
No strong opinion. I think we can keep it simple for now instead of walking through base types.